exercises/concept/windowing-system/.docs/instructions.md

```diff
@@ -26,60 +26,136 @@ screenSize.height ║                 |      │                      │
+The width is provided as the first parameter, the height as the second one.
+The default width and height should be `80` and `60`, respectively.
+
+Additionally, define a method `resize(newWidth, newHeight)` that takes a new width and height as parameters and changes the fields to reflect the new size.
+
+```javascript
+const size = new Size(1080, 764);
+size.width;
+// => 1080
+size.height;
+// => 764
+
+size.resize(1920, 1080);
+size.width;
+// => 1920
+size.height;
+// => 1080
 ```
 
-## 2. Define a Position struct
+## 2. Define Position to store a window position
+
+Define a class (constructor function) named `Position` with two fields, `x` and `y` that store the current horizontal and vertical position, respectively, of the window's upper left corner.
+The constructor function should accept initial values for these fields.
+The value for `x` is provided as the first parameter, the value for `y` as the second one.
+The default value should be `0` for both fields.
 
-Define a struct named `Position` with two `Int` properties, `x` and `y` that store the current horizontal and vertical position, respectively, of the window's upper left corner. The initial values of x and y should each be 0. The position (0, 0) is the upper left corner of the screen with `x` values getting larger as you move right and `y` values getting larger as you move down.
+The position (0, 0) is the upper left corner of the screen with `x` values getting larger as you move right and `y` values getting larger as you move down.
 
-Include a method `moveTo(newX:newY:)` that takes new x and y parameters and changes the properties to reflect the new position.
+Also define a method `moveTo(newX, newY)` that takes new x and y parameters and changes the properties to reflect the new position.
 
-```swift
-var point = Position(x: 10, y: 20)
-// => Position
-point.moveTo(newX: 100, newY: -100)
-point.y
-// => -100
+```javascript
+const point = new Position();
+point.x;
+// => 0
+point.y;
+// => 0
+
+point.moveTo(100, 200);
+point.x;
+// => 100
+point.y;
+// => 200
 ```
 
 ## 3. Define a Window class
 
-Define a window class with the following properties:
+Define a `Window` class with the following fields:
+
+- `screenSize`: holds a fixed value of type `Size` with `width` 800 and `height` 600
+- `size` : holds a value of type `Size`, the initial value is the default value of the `Size` instance
+- `position` : holds a value of type `Position`, the initial value is the default value of the `Position` instance
+
+The constructor should have no parameters.
 
-- `title` : `String`, Initial value is "New Window"
-- `screenSize` : `Size`, constant value with `width` = 800 and `height` = 600
-- `size` : `Size`, initial value is the default value of the `Size` struct
-- `position` : `Position`, initial value is the default value of the `Position` struct
-- `contents` : `String?`, initial value is `nil`
+```javascript
+const window = new Window();
```

Comment:
`Window` exists in the browser as class, so I recommend a slightly new name. `ProgramWindow` or `UIWindow` for example?

Reply:
Oh yes, good point! Will change that when I continue working on the exercise.